File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        ArrayList<Integer> unique = new ArrayList<>();
        
        int prev = nums[0];
        unique.add(nums[0]);
        for(int i=1; i<nums.length; i++){
            if( nums[i] != prev ){
                unique.add(nums[i]);
            }
            prev = nums[i];
        }
        int size = unique.size();
        for(int i=0; i<size; i++){
            nums[i] = unique.get(i);
        }
        return size;
    }
}

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.go

func removeDuplicates(nums []int) int {
    
    k := 0
    priv := nums[0] - 1
    priv_index := -1
    for i := 0; i<len(nums); i++ {
        if nums[i]> priv {
            priv_index++
            nums[priv_index] = nums[i]
            priv = nums[i]
            k++
        }
    }
    return k
}